### PR TITLE
dont cache tag results containing order:random

### PR DESCRIPTION
**Pull request by dogkisser**

### CODE REVIEW

**Comment by dogkisser** [comment]
i'd squash these but i can't figure out gcm

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -31,10 +36,13 @@ def get_tag_results(tag_string, after, before, link, limit = 15)
...
         return nil
       end
-
+  
       results = JSON.parse(response.body)['posts']
-
       if results.present? && results.class == Array
+        if /order:random/i !~ padded_tag_string
+          Rails.cache.write(key, results, expires_in: 45.minutes)
+        end
```
Comment (pupgray) [inline app/controllers/application_controller.rb:44]: I'd prefer to still cache random posts, if only for a short amount of time.

Reason being, a lot of forks have no changes in their blacklist from the parent, so a 1 minute cache would be ideal just to push thru any bot sets that hit a fork.

Last weeks botting mess saw me hammering e621 pretty badly, want to limit things where I can. (Expo backoff coming... uh... soon?)

```suggestion
        if /order:random/i =~ padded_tag_string
          Rails.cache.write(key, results, expires_in: 45.minutes)
        else
          Rails.cache.write(key, results, expires_in: 1.minute)
        end
```

**Review by pupgray — CHANGES_REQUESTED** [review]
LGTM, one nitpick

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (dogkisser) [reply app/controllers/application_controller.rb]: Could this situation ever arise? The only time `order:random` is legal is when it comes from porn bots, and they run every 10 minutes.
Better safe than banned I guess :3c

**Review by pupgray — APPROVED** [review]
lgtm!

**Pull request merged**